### PR TITLE
ECO-169 - Migrate nginx phase 2 - Remove ecomm app pages from nginx buildpack

### PR DESCRIPTION
**Pull request by tianheyang**
### Issue
https://linear.app/territory/issue/ECO-169/migrate-nginx-routing-to-cf-workers

### Summary
Removing the nginx buildback and moving routing rules over to CF workers will improve response times and make room for DataDog in the Rails app Heroku slug.

For phase 2 of this migration, the Heroku nginx build pack has been updated to remove proxying for the ecomm app, which is instead proxied by our gateway CF worker in the associated change [here](https://github.com/powersupplyhq/cloudflare-workers/pull/5).

### Screencast
Live logs for the staging2 CF worker, showing traffic to the blog being routed through the worker while running this updated buildpack on staging2:

https://user-images.githubusercontent.com/1920921/124223125-f3422200-dab7-11eb-9eda-84190c049525.mov

buildpack phase 2

### CODE REVIEW

File: config/nginx.conf.erb [hunk 1]
```diff
@@ -35,36 +35,11 @@ http {
...
     location /manage/ {
       resolver 1.1.1.1 valid=300s;
       resolver_timeout 10s;
       proxy_pass <%= ENV["ENGAGEMENT_APP_URL"] %>;
       proxy_redirect off;
     }
-
-    location ~ \/cart\/(plan|recipient) {
-      return 301 <%= "#{ENV['CHECKOUT_APP_SYNTH_URL']}plan?" %>$args;
-    }
```
Comment (tianheyang) [inline config/nginx.conf.erb:68]: All `/cart` routes should have been deprecated; I couldn't find anywhere we are still linking to these old funnel routes. So I believe these should be good to remove from the buildpack without needing to set up a 301 in the Rails app's `routes` file.

Comment (Aram-Anderson) [reply config/nginx.conf.erb:68]: The only reason this redirect was here was for any outstanding marketing links that might be out in the wild. But at this point, it's probably been long enough that we're clear.

**Review by Aram-Anderson — APPROVED** [review]
:shipit:

File: config/nginx.conf.erb (hunk 1 above)
Comment (mstoople532) [reply config/nginx.conf.erb:68]: idk I'd check in with @sophtown on this @tianheyang ... we've definitely been having issues with outstanding links and we may not want to drop support without checking

**Review by mstoople532 — APPROVED** [review]

File: config/nginx.conf.erb (hunk 1 above)
Comment (tianheyang) [reply config/nginx.conf.erb:68]: Great call; I will check in with @sophtown to make sure this is safe to remove. Thanks for the call-out @mstoople532 and context @Aram-Anderson ! 🙏

**Comment by tianheyang** [comment]
Merging this; confirmed with @sophtown that these are safe to remove. cc @jschne88 @Aram-Anderson @mstoople532

**Pull request merged**